graylog2-web-interface/src/components/events/EventsEntityTable.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React, { useMemo, useCallback } from 'react';
+
+import useCurrentUser from 'hooks/useCurrentUser';
+import useTableElements from 'components/events/useTableComponents';
+import CustomColumnRenderers from 'components/events/ColumnRenderers';
+import getStreamTableElements from 'components/events/Constants';
+import PaginatedEntityTable from 'components/common/PaginatedEntityTable';
+import FilterValueRenderers from 'components/events/FilterValueRenderers';
+import type { EventsAdditionalData } from 'components/events/fetchEvents';
+import fetchEvents, { keyFn } from 'components/events/fetchEvents';
+import type { SearchParams } from 'stores/PaginationTypes';
+import type { Event } from 'components/events/events/types';
+import type { PaginatedResponse } from 'components/common/PaginatedEntityTable/useFetchEntities';
+import useQuery from 'routing/useQuery';
+
+import QueryHelper from '../common/QueryHelper';
+
+const EventsEntityTable = () => {
+  const currentUser = useCurrentUser();
+  const { stream_id: streamId } = useQuery();
+
+  const columnRenderers = useMemo(() => CustomColumnRenderers(currentUser.permissions), [currentUser.permissions]);
```

Comment:
In similar cases we replaced the `CustomColumnRenderers` a `useColumnRenderes` hook, which can also include the `useMemo` and `useCurrentUser` part.